Services/UserService/Program.cs

```diff
@@ -1,36 +1,30 @@
 using System;
 using Microsoft.AspNetCore.DataProtection.KeyManagement;
+using Microsoft.EntityFrameworkCore;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
-using Supabase;
+using Npgsql;
+using UserService.Db;
+using UserService.Service;
 
 var builder = WebApplication.CreateBuilder(args);
 
 // Add services to the container.
-
 builder.Services.AddControllers();
 builder.Services.AddControllers().AddNewtonsoftJson();
 
 // Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
 builder.Services.AddEndpointsApiExplorer();
 builder.Services.AddSwaggerGen();
 
-// Supabase configuration
+// PostgreSQL configuration
 var configuration = builder.Configuration;
+var connectionString = configuration.GetConnectionString("DefaultConnection");
+builder.Services.AddDbContext<AppDbContext>(options => options.UseNpgsql(connectionString));
 
-var supabaseUrl = configuration["SupaBase:Url"];
-var supabaseKey = configuration["SupaBase:Key"];
-
-builder.Services.AddSingleton(_ =>
-{
-    var options = new Supabase.SupabaseOptions { AutoConnectRealtime = true };
-
-    var client = new Supabase.Client(supabaseUrl!, supabaseKey, options);
-    client.InitializeAsync().Wait();
-
-    return client;
-});
+// Dependency injection
+builder.Services.AddScoped(typeof(IDbService<>), typeof(DbService<>));
```

Comment:
You should not add service into the DI this way if the service dont use generics. Read about how to introduce scoped services.